pkg/roles/repository.go

```diff
@@ -0,0 +1,171 @@
+// Copyright 2025 Canonical Ltd.
+// SPDX-License-Identifier: AGPL-3.0
+
+package roles
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+
+	sq "github.com/Masterminds/squirrel"
+
+	"github.com/canonical/identity-platform-admin-ui/internal/logging"
+	"github.com/canonical/identity-platform-admin-ui/internal/monitoring"
+	"github.com/canonical/identity-platform-admin-ui/internal/tracing"
+	"github.com/canonical/identity-platform-admin-ui/pkg/storage"
+)
+
+type RoleRepository struct {
+	db storage.DBClientInterface
+
+	logger  logging.LoggerInterface
+	tracer  tracing.TracingInterface
+	monitor monitoring.MonitorInterface
+}
+
+func (r *RoleRepository) FindRole(ctx context.Context, userID, ID string) (*Role, error) {
+	ctx, span := r.tracer.Start(ctx, "roles.RoleRepository.GetRole")
+	defer span.End()
+
+	row := r.db.Statement().Select("id", "name").
+		From(storage.ROLE_TABLE).
+		Where(sq.Eq{storage.ROLE_ID: ID}).
+		Where(sq.Eq{storage.ROLE_OWNER: userID}).
+		QueryRowContext(ctx)
+
+	var id, name string
+
+	err := row.Scan(&id, &name)
+	if err != nil {
+		return nil, fmt.Errorf("unable to scan FindRole result, %w", err)
+	}
+
+	if id == "" {
+		return nil, storage.ErrNotFound
+	}
+
+	return &Role{
+		ID:   id,
+		Name: name,
+	}, nil
+}
+
+func (r *RoleRepository) ListRoles(ctx context.Context, userID string) ([]string, error) {
+	ctx, span := r.tracer.Start(ctx, "roles.RoleRepository.ListRoles")
+	defer span.End()
+
+	rows, err := r.db.Statement().Select("name").
+		From(storage.ROLE_TABLE).
+		Where(sq.Eq{storage.ROLE_OWNER: userID}).
+		QueryContext(ctx)
+
+	if errors.Is(err, sql.ErrNoRows) {
+		return []string{}, nil
+	}
+
+	if err != nil {
+		return nil, fmt.Errorf("unable to list roles, %w", err)
+	}
+
+	defer func() { _ = rows.Close() }()
```

Comment:
(probably stupid) question: can't we re-write this to
```suggestion
	defer rows.Close()
```
?